packages/wrangler/src/bundle.ts

```diff
@@ -98,28 +106,47 @@ export async function bundleWorker(
 	// `checked-fetch.js` to do so. However, with yarn 3 style pnp,
 	// we need to extract that file to an accessible place before injecting
 	// it in, hence this code here.
-	const osTempDir = os.tmpdir();
-	const checkedFetchFileToInject = path.join(
-		osTempDir,
-		"--temp-wrangler-files--",
-		"checked-fetch.js"
-	);
+
+	const checkedFetchFileToInject = path.join(tmpDir.path, "checked-fetch.js");
 
 	if (checkFetch && !fs.existsSync(checkedFetchFileToInject)) {
-		fs.mkdirSync(path.join(osTempDir, "--temp-wrangler-files--"), {
+		fs.mkdirSync(tmpDir.path, {
 			recursive: true,
 		});
 		fs.writeFileSync(
 			checkedFetchFileToInject,
 			fs.readFileSync(path.resolve(__dirname, "../templates/checked-fetch.js"))
 		);
 	}
-	// TODO: we need to have similar logic like above for other files
-	// like the static asset facade, and other middleware that we
-	// plan on injecting/referencing.
+
+	// At this point, we take the opportunity to "wrap" any input workers
+	// with any extra functionality we may want to add. This is done by
+	// passing the entry point through a pipeline of functions that return
+	// a new entry point, that we call "middleware" or "facades".
+	// Look at implementations of these functions to learn more.
+
+	type MiddlewareFn = (arg0: Entry) => Promise<Entry>;
+	const middleware: (false | MiddlewareFn)[] = [
+		serveAssetsFromWorker &&
+			((currentEntry: Entry) => {
+				return applyStaticAssetFacade(currentEntry, tmpDir.path);
+			}),
+		formatErrorsInDev &&
+			((currentEntry: Entry) => {
+				return applyFormatDevErrorsFacade(currentEntry, tmpDir.path);
+			}),
+	].filter((x) => x !== false);
+
+	let inputEntry = entry;
+
+	for (const middlewareFn of middleware as MiddlewareFn[]) {
+		inputEntry = await middlewareFn(inputEntry);
+	}
+
+	// At this point, inputEntry points to the entry point we want to build.
 
 	const result = await esbuild.build({
-		...getEntryPoint(entry.file, serveAssetsFromWorker),
+		entryPoints: [inputEntry.file],
```

Comment:
Is this collecting all the inputEntry's or is this esbuild.build() only called on the last one?

Reply:
Just the last one. It's each individual middleware's responsibility to point to the next one in the chain (Look for `__ENTRY_POINT__`)